### PR TITLE
Added most of csvkit

### PR DESCRIPTION
**Pull request by oxguy3**
Added most of the tools from [csvkit](https://csvkit.readthedocs.org), a command-line toolbelt for dealing with CSVs and other tabular data formats. Specifically, I added:

* csvclean
* csvcut
* csvformat
* csvgrep
* csvlook
* csvpy
* csvsort
* csvstat
* in2csv

(still missing from csvkit: csvstack, csvjson, csvsql, sql2csv)

### CODE REVIEW

File: pages/common/csvlook.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# csvlook
+
+> Render a CSV file in the console as a fixed-width table.
+> Included in csvkit.
+
+- View a CSV file:
+
+`csvlook {{data.csv}}`
+
+- View a CSV file with _less_ for easy scrolling:
```
Comment (rubenvereecken) [inline pages/common/csvlook.md]: I realize this is a common example, but it doesn't really belong to this page as this one is more about `less` than `csvlook`.

Comment (oxguy3) [inline pages/common/csvlook.md]: True. csvlook really doesn't do anything besides the most basic first example. Should I cut it down to just that (or just remove it entirely even)?

File: pages/common/csvpy.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# csvpy
+
+> Loads a CSV file into a Python shell.
```
Comment (rubenvereecken) [inline pages/common/csvpy.md]: To anyone else mystified by what this does: it drops in a shell and the prints a friendly message saying you can access your file through the `reader` variable. Self-explanatory enough once you try it out.

File: pages/common/csvsort.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+- Sort a CSV file by column 9:
+
+`csvsort -c {{9}} {{data.csv}}`
+
+- Sort a CSV file by the "name" column in descending order:
+
+`csvsort -r -c {{name}} {{data.csv}}`
+
+- Sort a CSV file by columns 2, then by column 4:
```
Comment (rubenvereecken) [inline pages/common/csvsort.md]: _column_ 2

File: pages/common/csvsort.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+`csvsort -c {{2,4}} {{data.csv}}`
+
+- Sort a CSV file without inferring data types:
```
Comment (rubenvereecken) [inline pages/common/csvsort.md]: Basically a string sort? Is this ever particularly useful? I wonder when you used it yourself, otherwise you probably wouldn't have added it.

File: pages/common/csvstat.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+`csvstat --sum {{data.csv}}`
+
+- Show averages for all columns:
+
+`csvstat --mean {{data.csv}}`
+
+- Show the max value length for column 3:
+
+`csvstat -c {{1}} --len {{data.csv}}`
```
Comment (rubenvereecken) [inline pages/common/csvstat.md]: You sure ;)?

File: pages/common/csvstat.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
+# csvstat
```
Comment (rubenvereecken) [inline pages/common/csvstat.md]: @igorshubovych is this one too long?

File: pages/common/csvpy.md (hunk 2 above)
Comment (oxguy3) [inline pages/common/csvpy.md]: Yeah, I struggled a bit trying to find an explanation of what this does that wasn't too wordy. :stuck_out_tongue:

**Comment by rubenvereecken** [comment]
Question by the way, maybe you can help me out. You look like you know your way around GitHub, you fixed the previous PR when Travis complained. The linter we use in Travis can actually automatically format pages so people don't have to bother, but I haven't managed to get the word out very well so far. Any ideas where you would have expected this type of information?

We have the linter in the git pre-commit hooks I believe, once installed through `make setup` or something. Still, that formatter could save people a world of pain.

File: pages/common/csvsort.md (hunk 4 above)
Comment (oxguy3) [inline pages/common/csvsort.md]: It comes in handy from time to time, especially if you want to pass the CSV off to someone else (particularly a non-developer) or publish it or something.

File: pages/common/csvlook.md (hunk 1 above)
Comment (rubenvereecken) [inline pages/common/csvlook.md]: I guess we can cut it down to just the trivial example. It doesn't hurt to have the page, however simple.

**Comment by oxguy3** [comment]
Oh, awesome -- I figured there probably was a better way than just failing over and over. I probably would have noticed you guys have that if it were mentioned in CONTRIBUTING.md. I'll check that out for sure.

**Comment by rubenvereecken** [comment]
It's in there, just fairly hidden I guess. I'll try to put it in or near the TLDR somewhere, flag it for attention somehow. Thanks!

**Comment by oxguy3** [comment]
Ah, whoops, I'm blind. I guess I sorta skimmed past the "submitting a pull request" section because I already knew how to use git. That's on me; I should have definitely read all that. Mea culpa! >.<

**Comment by rubenvereecken** [comment]
No man it's exactly what I would have done and I feared this would be the case. I added it (#704) in more plain view right next to our syntax, I hope it'll jump out better this way. Thanks for confirming my suspicions :).

File: pages/common/csvstat.md (hunk 6 above)
Comment (oxguy3) [inline pages/common/csvstat.md]: I just removed one of the examples (the sums and means commands work exactly the same, so I removed the latter)

**Comment by rubenvereecken** [comment]
LGTM!

File: pages/common/in2csv.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+- Convert a DBF file to a CSV file:
+
+`in2csv {{data.dbf}} > {{data.csv}}`
+
+- Convert a specific sheet from an XLSX file to CSV:
+
+`in2csv --sheet={{sheet_name}} {{data.xlsx}}`
+
+- Fetch csvkit's open issues from GitHub's JSON API, and convert them to CSV:
```
Comment (rubenvereecken) [inline pages/common/in2csv.md]: Actually I missed this one. That's a really cool oneliner and demonstrates the real usefulness of the tool but it's a little too much beyond the tool's core. I sometimes don't mind very practical examples but it's not completely in the tldr spirit, so I'll ask @igorshubovych to have a second look.

Comment (igorshubovych) [inline pages/common/in2csv.md]: I agree with @rubenvereecken.

If you really want to demonstrate the usage of `-f`, then you would better simplify it to something like this:
```md
`cat issues.json | in2csv -f json > issues.csv`
```

Comment (rubenvereecken) [inline pages/common/in2csv.md]: I'd second that, we have some more like this one. `cat` is nice and basic. It might even be possible to drop the stream to file since a reader should well realize that csv is coming out, ready to be written anywhere.

File: pages/common/csvclean.md [hunk 8]
```diff
@@ -0,0 +1,12 @@
+# csvclean
+
+> Finds and cleans common syntax errors in CSV files.
+> Included in csvkit.
```
Comment (waldyrious) [inline pages/common/csvclean.md]: This may be misread as "including [errors] in csvkit." by an inattentive reader (e.g. me). How about making it explicit, i.e. "This command is included in csvkit"? Or is that overkill?

File: pages/common/csvcut.md [hunk 9]
```diff
@@ -0,0 +1,20 @@
+# csvcut
+
+> Filter and truncate CSV files. Like unix "cut" command, but for tabular data.
```
Comment (waldyrious) [inline pages/common/csvcut.md]: Capitalize "Unix", and either write "Unix's cut command", or "the Unix cut command" (I prefer the former). Also, it would probably be more appropriate to format "cut" with backticks, rather than quote marks.

File: pages/common/csvcut.md [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+
+- Print indices and names of all columns:
+
+`csvcut -n {{data.csv}}`
+
+- Extract the first and third columns:
+
+`csvcut -c {{1,3}} {{data.csv}}`
+
+- Extract all columns EXCEPT the fourth one:
```
Comment (waldyrious) [inline pages/common/csvcut.md]: Can't we use markdown formatting here, instead of all caps? @igorshubovych

File: pages/common/csvcut.md [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+
+`csvcut -C {{4}} {{data.csv}}`
+
+- Extract the columns named "id" and "first name" (in that order):
+
+`csvcut -c {{id,"first name"}} {{data.csv}}`
```
Comment (waldyrious) [inline pages/common/csvcut.md]: Huh. What happens if, say, the fourth column's title is literally "3"?

File: pages/common/csvformat.md [hunk 12]
```diff
@@ -0,0 +1,24 @@
+# csvformat
+
+> Convert a CSV file to a custom output format.
+> Included in csvkit.
+
+- Convert to a tab-delimited file:
```
Comment (waldyrious) [inline pages/common/csvformat.md]: Perhaps include "(tsv)" in the end, just to be sure people get this quickly?

File: pages/common/csvformat.md [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+
+`csvformat -T {{data.csv}}`
+
+- Convert delimiters to a custom character:
+
+`csvformat -D "{{custom_character}}" {{data.csv}}`
+
+- Convert line endings to carriage return + line feed:
+
+`csvformat -M "{{\r\n}}" {{data.csv}}`
```
Comment (waldyrious) [inline pages/common/csvformat.md]: Does "M" stand for anything here?...

File: pages/common/csvformat.md [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+
+- Convert to minimal use of quote characters:
```
Comment (waldyrious) [inline pages/common/csvformat.md]: Maybe "Convert to minimal use" --> "Minimize use", and append "(omit them wherever unnecessary)" or something else that explains what it does?

File: pages/common/csvformat.md [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+
+`csvformat -U 0 {{data.csv}}`
+
+- Convert to maximum use of quote characters:
```
Comment (waldyrious) [inline pages/common/csvformat.md]: Same here: "Maximize use..." and an explicit parenthetical note -- "wrap all fields in quote characters", or something like that.

File: pages/common/csvpy.md (hunk 2 above)
Comment (waldyrious) [inline pages/common/csvpy.md]: I'll be honest, I still don't get it even after reading @rubenvereecken's explanation. Maybe I should try it out :P

File: pages/common/csvpy.md [hunk 16]
```diff
@@ -0,0 +1,12 @@
...
+> Included in csvkit.
+
+- Load a CSV file into a _CSVKitReader_ object:
```
Comment (waldyrious) [inline pages/common/csvpy.md]: Is this meant to be markdown formatting? Because github certainly seems to be formatting it as italics.

**Comment by waldyrious** [comment]
Ok this is a massive (by our standards) contribution -- thanks a lot, @oxguy3! Don't let my inline comments discourage you, this looks OK to merge as-is if you are tired of our pickiness :P, but I'd like your replies anyway, in order to edit the page later.

File: pages/common/csvformat.md (hunk 13 above)
Comment (rubenvereecken) [inline pages/common/csvformat.md]: Just a guess that this might stand for ^M, the carriage return?

File: pages/common/csvpy.md (hunk 16 above)
Comment (rubenvereecken) [inline pages/common/csvpy.md]: That's Markdown. I concur. Generally we tend to use backticks for formatting or referring to tokens or the like @oxguy3

File: pages/common/csvformat.md (hunk 13 above)
Comment (waldyrious) [inline pages/common/csvformat.md]: Good point -- we could add that to the description, e.g. "carriage return (^M) + line feed" even if that's not the original meaning, as it serves as a nice mnemonic :)

**Comment by igorshubovych** [comment]
@oxguy3, will you address @waldyrious comments?

Just FYI, it can be merged as is, the pages are already good. However some of @waldyrious comments make sense.

**Comment by waldyrious** [comment]
Only some @igorshubovych? :stuck_out_tongue_winking_eye:

**Comment by igorshubovych** [comment]
I want them all

**Comment by rubenvereecken** [comment]
This is our contributor team at its best.

File: pages/common/csvformat.md (hunk 15 above)
Comment (oxguy3) [inline pages/common/csvformat.md]: You don't have to wrap all fields in quote characters. If they have spaces, you need to use quotes or escape the spaces. It's the same as how any other *nix command handles spaces, so I didn't think it needed an extra explanation.

File: pages/common/csvcut.md (hunk 11 above)
Comment (oxguy3) [inline pages/common/csvcut.md]: Not sure, to be honest, and the documentation doesn't clarify. I think in general, if you have columns named with numbers, it's your own dang fault if you end up with confusing behavior. :stuck_out_tongue:

File: pages/common/csvclean.md (hunk 8 above)
Comment (oxguy3) [inline pages/common/csvclean.md]: Hmm, I'm not sure. I'm tempted to stick with the less wordy options because, by Markdown standards, these two separate lines actually combine into one long line.

**Comment by oxguy3** [comment]
Hehe, sorry for the slow responses to the comments. I've either fixed or responded to all of them now!

File: pages/common/csvclean.md (hunk 8 above)
Comment (waldyrious) [inline pages/common/csvclean.md]: Fair enough. It's not a big deal anyway.

File: pages/common/csvcut.md (hunk 11 above)
Comment (waldyrious) [inline pages/common/csvcut.md]: True.

File: pages/common/csvformat.md (hunk 15 above)
Comment (waldyrious) [inline pages/common/csvformat.md]: I'm not sure I get what's the difference then, and what "maximum" and "minimum" mean here.

**Comment by rubenvereecken** [comment]
I'm merging in this absolute beauty. Thanks for all the effort @oxguy3 and thanks for the discussion @waldyrious. Whoop :tada:

**Comment by oxguy3** [comment]
Woohoo! :fireworks:

**Comment by igorshubovych** [comment]
@rubenvereecken thanks, for doing my job.

**Comment by rubenvereecken** [comment]
_gasps_

My bad.
